thinc/backends/ops.py

```diff
@@ -351,23 +357,23 @@ def list2padded(self, seqs: List[Floats2d]) -> Padded:
             batch_size_at_t_[t] = current_size
         assert sum(lengths_) == sum(batch_size_at_t_)
         return Padded(
-            cast(Floats3d, arr),
+            arr,
             self.asarray1i(batch_size_at_t_),
             self.asarray1i(lengths_),
             self.asarray1i(indices_),
         )
 
-    def padded2list(self, padded: Padded) -> List2d:
+    def padded2list(self, padded: Padded) -> List[Array2d]:
```

Comment:
Similar to earlier comments, I don't think we want to widen the scope to support lists of mixed ints & floats, so I think the original annotation was actually fine? I think it would be even more clear to use `ArrayT2d_s` again.

Reply:
That makes sense, but in practice if you make that change the code breaks in a large number of cases in situations where the calling class doesn't know which of the two types of list — `int`s or `float`s — it has. Because you can't use strict typing for the reason discussed above, the only alternative is to force all calling classes to cast to `List2d` even if they do know that they are passing only `Float`s or only `Int`s.

Reply:
I'm not sure I follow because the type under consideration is not a parameter being passed by a calling class, but rather the return type here. Can you give an example of calling code that would break?

Reply:
e.g. the layer `padded2list` which is used in various places, accepts a `Padded` that can contain either `Floats` or `Ints` but there is no way of knowing which. Generally if you specify a strict type somewhere, the type trickles through the code until some such call is required, if not in the forward pass in the backprop.